Remotion/Web/ClientScript/Scripts/SmartPage.ts

```diff
@@ -498,15 +504,60 @@ class SmartPage_Context
   // Marks the page as loaded.
   public SetCacheDetectionFieldLoaded(): void
   {
-    window.sessionStorage.setItem(this.GetPageToken(), SmartPage_CacheDetection[SmartPage_CacheDetection.HasLoaded]);
+    this.SetCacheDetectionField(SmartPage_CacheDetection.HasLoaded);
   };
 
   // Marks the page as submitted.
   public SetCacheDetectionFieldSubmitted(): void
   {
-    window.sessionStorage.setItem(this.GetPageToken(), SmartPage_CacheDetection[SmartPage_CacheDetection.HasSubmitted]);
+    this.SetCacheDetectionField(SmartPage_CacheDetection.HasSubmitted);
   };
 
+  private SetCacheDetectionField(cacheDetectionValue: SmartPage_CacheDetection): void
+  {
+    const tokenKey = this._pageTokenSessionStoragePrefix + this.GetPageToken();
+
+    if (window.sessionStorage.getItem(tokenKey) === null)
+      this.IncreasePageTokenSessionStorageEntryCount();
+
+    window.sessionStorage.setItem(tokenKey, SmartPage_CacheDetection[cacheDetectionValue]);
+    this.ClearOldPageTokenSessionStorageEntries();
```

Comment:
```suggestion
    if (window.sessionStorage.getItem(tokenKey) === null)
    {
      this.IncreasePageTokenSessionStorageEntryCount();
      this.ClearOldPageTokenSessionStorageEntries();
    }

    window.sessionStorage.setItem(tokenKey, SmartPage_CacheDetection[cacheDetectionValue]);
```

Just for consideration - if we don't add something, we don't need to perform any cleanup. I know the count already optimizes things, but it may also make things more expressive this way. Thoughts?

Reply:
I thought about that but somehow it felt wrong to increase the counter and potentially trigger a cleanup without the item actually being added at this point, that's why I chose the existing path (and order of execution). What do you think about this?

Reply:
also considered the same reasoning. Landed on the other side because a) it's just 1 item in a list of 100, so the off by one "error" is not relevant for what we're trying to achieve. In the end, for me it came down to reading this method and thinking, that if I just do an update, not an insert, doing "always" a cleanup (without considereing the also implemented performance improvement) felt less optimized in this method.

But it really is a matter of preference while trying to find the local code optimum until the one extra lookup of the counter becomes a performance bottleneck (so, never?)